mica/starcheck/starcheck.py

```diff
@@ -20,60 +22,197 @@
 
 DEFAULT_CONFIG = dict(dbi='sqlite',
                       server=os.path.join(MICA_ARCHIVE, 'starcheck', 'starcheck.db3'),
-                      mp_top_level='/data/mpcrit1/mplogs')
+                      mp_top_level='/data/mpcrit1/mplogs',
+                      timelines_db=dict(dbi='sqlite',
+                                        server=os.path.join(os.environ['SKA'], 'data', 'cmd_states', 'cmd_states.db3')))
 FILES = dict(data_root=os.path.join(MICA_ARCHIVE, 'starcheck'),
              touch_file=os.path.join(MICA_ARCHIVE, 'starcheck', "starcheck_parser.touch"),
              sql_def='starcheck.sql')
 
 # this doesn't belong in starcheck
-def get_timeline_at_date(date):
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    return aca_db.fetchone(
+def get_timeline_at_date(date, timelines_db=None):
+    date = DateTime(date).date
+    if timelines_db is None:
+        timelines_db = Ska.DBI.DBI(**DEFAULT_CONFIG['timelines_db'])
+    return timelines_db.fetchone(
         "select * from timeline_loads where datestop >= '%s' "
         " and datestart <= '%s' and scs <= 130 order by datestart desc"
         % (date, date))
 
-def get_mp_dir(obsid, config=None):
+
+def get_flickpix_mons(start=None, config=None):
     if config is None:
         config = DEFAULT_CONFIG
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    possible_runs = aca_db.fetchall(
-        "select * from tl_obsids where obsid = %d" % obsid)
-    if not len(possible_runs):
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
-        sc = db.fetchall(
+
+    # I'm ignoring the weird case of SIR obsids or whatever where the following ER keeps
+    # the MON
+    start_date = DateTime(start or '1999:001').date
+    with Ska.DBI.DBI(dbi='sqlite', server=config['server']) as db:
+        mons = db.fetchall("""select obsid, mp_starcat_time as mp_starcat_date, type, sz, yang, zang, dir
+                          from starcheck_catalog, starcheck_id
+                          where type = 'MON' and obsid > 40000
+                          and mp_starcat_date >= '{}'
+                          and starcheck_id.id = starcheck_catalog.sc_id""".format(start_date))
+        mons = Table(mons)
+    with Ska.DBI.DBI(**config['timelines_db']) as timelines_db:
+        statuses = []
+        # Check which ones actually ran or are likely to run
+        for mon in mons:
+            try:
+                catalog = get_starcheck_catalog_at_date(mon['mp_starcat_date'],
```

Comment:
A bit confused by the implementation suggestion here.  Did you mean `mons[i_mon]['catalog'] = catalog` or are you saying you want me to replace the entries of mons with catalogs?  And if you meant `mons[i_mon]['catalog'] = catalog` why would I need to use `i_mon` to index to the row instead of just making the assignment in the row as `mon['catalog'] = catalog`?